src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs

```diff
@@ -67,6 +67,17 @@ public void Initialize()
                             .Select(d => this.dBreezeSerializer.Deserialize<Poll>(d.Value))
                             .ToArray();
 
+                        // If the polls repository contains duplicate polls then reset and rebuild it.
+                        var uniquePolls = new HashSet<Poll>(polls);
+                        if (uniquePolls.Count != polls.Length)
+                        {
+                            this.logger.LogWarning("The polls repo contains {0} duplicate polls. Will rebuild it.", polls.Length - uniquePolls.Count);
+
+                            this.ResetLocked(transaction);
+                            transaction.Commit();
+                            return;
```

Comment:
If we return here, shouldnt we first set the `highestPollId` first?

Reply:
I think the tests that are failing in the rewind PR is because of the above:

https://dev.azure.com/StratisProject/StratisFullNode/_build/results?buildId=29444&view=results

Reply:
Its set in `ResetLocked`. That seems to be an old failure.